lsp-mode.el

```diff
@@ -5209,12 +5245,31 @@ It will show up only if current point has signature help."
 
 (defun lsp-execute-code-action-by-kind (command-kind)
   "Execute code action by name."
-  (if-let ((action (->> (lsp-get-or-calculate-code-actions command-kind)
-                        (-filter (-lambda ((&CodeAction :kind?))
-                                   (and kind? (equal command-kind kind?))))
-                        lsp--select-action)))
-      (lsp-execute-code-action action)
-    (signal 'lsp-no-code-actions '(command-kind))))
+  (->> (lsp-get-or-calculate-code-actions command-kind)
+       (-filter (-lambda ((&CodeAction :kind?))
+                  (and kind? (equal command-kind kind?))))
+       lsp--select-enabled-action
```

Comment:
If I call explicitly a refactoring and the method is disabled I would like to see the reason for which it is disabled. E. g. if I call "Extract method" I might see an error "You have selected a range that cannot be extracted to a method".

Reply:
Fixed. There might be a prompt with the reasons listed individually, depending on the number of actions and `lsp-auto-execute-action`.